Emby.Server.Implementations/Library/UserManager.cs

```diff
@@ -1,1223 +1,1222 @@
+
+        #region UserDeleted Event
+        /// <summary>
+        /// Occurs when [user deleted].
+        /// </summary>
+        public event EventHandler<GenericEventArgs<User>> UserDeleted;
+        /// <summary>
+        /// Called when [user deleted].
+        /// </summary>
+        /// <param name="user">The user.</param>
+        private void OnUserDeleted(User user)
+        {
+            UserDeleted?.Invoke(this, new GenericEventArgs<User> { Argument = user });
+        }
+        #endregion
+
+        /// <summary>
+        /// Gets a User by Id
+        /// </summary>
+        /// <param name="id">The id.</param>
+        /// <returns>User.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public User GetUserById(Guid id)
+        {
+            if (id.Equals(Guid.Empty))
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            return Users.FirstOrDefault(u => u.Id == id);
+        }
+
+        /// <summary>
+        /// Gets the user by identifier.
+        /// </summary>
+        /// <param name="id">The identifier.</param>
+        /// <returns>User.</returns>
+        public User GetUserById(string id)
+        {
+            return GetUserById(new Guid(id));
+        }
+
+        public User GetUserByName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            return Users.FirstOrDefault(u => string.Equals(u.Name, name, StringComparison.OrdinalIgnoreCase));
+        }
+
+        public void Initialize()
+        {
+            _users = LoadUsers();
+
+            var users = Users.ToList();
+
+            // If there are no local users with admin rights, make them all admins
+            if (!users.Any(i => i.Policy.IsAdministrator))
+            {
+                foreach (var user in users)
+                {
+                    user.Policy.IsAdministrator = true;
+                    UpdateUserPolicy(user, user.Policy, false);
+                }
+            }
+        }
+
+        public bool IsValidUsername(string username)
+        {
+            //The old way was dumb, we should make it less dumb, lets do so.
+            //This is some regex that matches only on unicode "word" characters, as well as -, _ and @
+            //In theory this will cut out most if not all 'control' characters which should help minimize any weirdness
+            string UserNameRegex = "^[\\w-'._@]*$";
+            // Usernames can contain letters (a-z + whatever else unicode is cool with), numbers (0-9), dashes (-), underscores (_), apostrophes ('), and periods (.)
+            return Regex.IsMatch(username, UserNameRegex);
+        }
+
+        private static bool IsValidUsernameCharacter(char i)
+        {
+            string UserNameRegex = "^[\\w-'._@]*$";
+            return Regex.IsMatch(i.ToString(), UserNameRegex);

```

Comment:
don't repeat this much, just
```suggestion
            return IsValidUsername(i.ToString());
```

Reply:
I had to make IsValidUsername static, but there was no good reason for it not to be anyhow so this fix is implemented.